discovery/digitalocean/digitalocean.go

```diff
@@ -0,0 +1,224 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package digitalocean
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/digitalocean/godo"
+	"github.com/go-kit/kit/log"
+	conntrack "github.com/mwitkow/go-conntrack"
+	config_util "github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/common/version"
+
+	"github.com/prometheus/prometheus/discovery/refresh"
+	"github.com/prometheus/prometheus/discovery/targetgroup"
+)
+
+const (
+	doLabel            = model.MetaLabelPrefix + "do_"
+	doLabelID          = doLabel + "droplet_id"
+	doLabelName        = doLabel + "droplet_name"
+	doLabelImage       = doLabel + "image"
+	doLabelPrivateIPv4 = doLabel + "private_ipv4"
+	doLabelPublicIPv4  = doLabel + "public_ipv4"
+	doLabelPublicIPv6  = doLabel + "public_ipv6"
+	doLabelRegion      = doLabel + "region"
+	doLabelSize        = doLabel + "size"
+	doLabelStatus      = doLabel + "status"
+	doLabelFeatures    = doLabel + "features"
+	doLabelTags        = doLabel + "tags"
+	separator          = ","
+)
+
+// DefaultSDConfig is the default DigitalOcean SD configuration.
+var DefaultSDConfig = SDConfig{
+	Port:            80,
+	RefreshInterval: model.Duration(60 * time.Second),
+}
+
+// SDConfig is the configuration for DO based service discovery.
+type SDConfig struct {
+	BearerToken     config_util.Secret `yaml:"bearer_token,omitempty"`
```

Comment:
Can we not use standard http config here, rather than partially reinventing it and losing things like tls/proxy_url?

Reply:
done